### PR TITLE
Node mode controllers

### PR DESCRIPTION
**Pull request by gambol99**
- Changing the `controllers` flag to a `[]string`, not the last PR wouldn't have worked as the `flagbuilder` does not handle pointers to a string slice.
- Defaulting to adding the tokencleaner if the bootstrap tokens are enabled

### CODE REVIEW

File: pkg/model/components/kubecontrollermanager.go [hunk 1]
```diff
@@ -167,5 +167,13 @@ func (b *KubeControllerManagerOptionsBuilder) BuildOptions(o interface{}) error
 		}
 	}
 
+	// @check if the node authorization is enabled and if so enable the tokencleaner controller (disabled by default)
+	// This is responsible for cleaning up bootstrap tokens which have expired
+	if b.Context.IsKubernetesGTE("1.10") {
+		if fi.BoolValue(clusterSpec.KubeAPIServer.EnableBootstrapAuthToken) && len(kcm.Controllers) <= 0 {
```
Comment (KashifSaadat) [inline pkg/model/components/kubecontrollermanager.go]: If a user has enabled bootstrap tokens for their cluster and provides a list of KCM controllers (excluding the tokencleaner), should we still append to the list here, or just take their literal set of controllers to apply?

**Comment by KashifSaadat** [comment]
Ah sorry I missed this in the last PR! Looks good, just one minor comment and will wait for tests to finish. :)

File: pkg/model/components/kubecontrollermanager.go (hunk 1 above)
Comment (gambol99) [reply pkg/model/components/kubecontrollermanager.go]:
```shell
A list of controllers to enable. '*' enables all on-by-default controllers, 'foo' enables the controller named 'foo', '-foo' disables the controller named 'foo'.
All controllers: attachdetach, bootstrapsigner, clusterrole-aggregation, cronjob, csrapproving, csrcleaner, csrsigning, daemonset, deployment, disruption, endpoint, garbagecollector, horizontalpodautoscaling, job, namespace, nodeipam, nodelifecycle, persistentvolume-binder, persistentvolume-expander, podgc, pv-protection, pvc-protection, replicaset, replicationcontroller, resourcequota, route, service, serviceaccount, serviceaccount-token, statefulset, tokencleaner, ttl, ttl-after-finished
Disabled-by-default controllers: bootstrapsigner, tokencleaner
```
So only two controllers are actually disabled by default .. I'm going on the basis that if they are setting the `controllers` flag they know what they are doing, or at the very least taking responsibility for it ..

Comment (chrisz100) [reply pkg/model/components/kubecontrollermanager.go]: If we do that: documentation should be very clear on what and how this is going to affect the gs. Also: needs to be backwards compatible to not break things as “know what you are doing” is good, but so many people implement it in pipelines and scripts - especially if they get this deep into the parameters.

**Comment by KashifSaadat** [comment]
Once the NodeAuthorizer / Bootstrap tokens work is moved to a more stable release within kops, will need to make sure enabling the KCM tokencleaner controller is highlighted in the docs.

**Pull request merged**